subsys/bluetooth/audio/stream.c

```diff
@@ -477,7 +477,7 @@ static void audio_stream_qos_cleanup(const struct bt_conn *conn,
 	struct bt_audio_stream *stream;
 
 	SYS_SLIST_FOR_EACH_CONTAINER(&group->streams, stream, _node) {
-		if (stream->conn != conn && stream->ep != NULL) {
+		if (stream->conn != conn || stream->ep == NULL) {
```

Comment:
This change does somewhat make sense, but currently in the code there is no place where `stream->ep` will be NULL when this function is called. 

Can you please elaborate on what issue you are attempting to fix?

Reply:
Warnings with static tool. If `stream->ep` cannot be NULL the NULL check should be removed.

Reply:
Which tool did you use, and can you post the warning here?? Just so we have a reference for the future :)

Anyways, we could replace the `stream->ep != NULL` with an `__ASSERT(stream->ep != NULL` so that the warning gets fixed, but we still have some check here in case that we would otherwise introduce a bug somewhere. 

Also please remove your first commit in this PR, as that isn't a valid change.

Reply:
The warning is obvious, if there is NULL check then it can be NULL. If it cannot be NULL please remove NULL check.
The same applies to the first patch.

Reply:
The reason why I am asking for the tool and warning, is that there are so many static code analyzers out there, and some of they are even mutually exclusive (i.e. some require changes that would make others fail), and thus we cannot adhere to all of them.